hpx/runtime/components/server/executor_component.hpp

```diff
@@ -68,10 +69,16 @@ namespace hpx { namespace components
                     &executor_component::execute, desc.get_description()),
                 std::move(data.func));
         }
+#endif
 
         template <typename Executor_ = Executor>
         static typename std::enable_if<
-            !traits::is_threads_executor<Executor_>::value>::type
+#if defined(HPX_HAVE_THREAD_EXECUTORS_COMPATIBILITY)
+            !traits::is_threads_executor<Executor_>::value
+#else
+            true
+#endif
```

Comment:
This will not work properly as SFINAE needs to rely on a 'local' template argument.

Reply:
Can you explain in what sense it won't work? Seeing this again I would perhaps `ifdef` out the whole `enable_if` when the compatibility option isn't defined. When it's off this is the only possible candidate.

Reply:
Yes, `#ifdef`ing the whole `enable_if` should be the way to go.